src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import StyleObjectPropType from '../../prop-types/style';
+import TabsNav from './TabsNav';
+import TabsPanes from './TabsPanes';
+import TabsPane from './TabsPane';
+import TabsTab from './TabsTab';
+
+let lastInstanceId = 0;
+
+const propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    selectedIndex: PropTypes.number,
+    style: StyleObjectPropType(),
+    tabsVisible: PropTypes.number,
+};
+
+const defaultProps = {
+    children: null,
+    className: null,
+    selectedIndex: 0,
+    style: null,
+    tabsVisible: null,
+};
+
+class Tabs extends Component {
+    constructor(props) {
+        super(props);
+
+        lastInstanceId += 1;
+
+        this.state = {
+            selectedIndex: this.props.selectedIndex,
+        };
+    }
+
+    render() {
+        const children = React.Children.map(this.props.children, (child) => {
```

Comment:
I wonder if we should split this into multiple functions to make it easier to read and maintain?
Perhaps:
`getChildren()`
Which calls:
`getTabsNav()`
`getTabsPane()`

Reply:
Changed